src/dali/plugin/input/csv/coinbase.py

```diff
@@ -0,0 +1,206 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# CSV Format: Transaction ID, Transaction Type, Date & time, Asset Acquired, "Quantity Acquired (Bought, Received, etc)" ,Cost Basis (incl. fees paid) (USD) ,Data Source   ,"Asset Disposed (Sold, Sent, etc)" ,Quantity Disposed ,Proceeds (excl. fees paid) (USD)
+
+import logging
+from csv import reader
+from typing import Dict, List, Optional
+from decimal import Decimal
+
+from rp2.logger import create_logger
+from rp2.rp2_decimal import RP2Decimal
+from dali import transaction_resolver
+
+from dali.abstract_input_plugin import AbstractInputPlugin
+from dali.abstract_transaction import AbstractTransaction
+from dali.configuration import Keyword
+from dali.in_transaction import InTransaction
+from dali.intra_transaction import IntraTransaction
+from dali.out_transaction import OutTransaction
+
+# transaction types
+_BUY = "Buy"
+_SEND = "Send"
+_REWARD = "Reward"
+_INCOMING = "Incoming"
+_FORK = "Fork"
+_AIRDROP = "Airdrop"
+_CONVERT = "Convert"
+
+
+class InputPlugin(AbstractInputPlugin):
+
+    __COINBASE: str = "Coinbase"
+
+    __TRANSACTION_ID_INDEX = 0
+    __TRANSACTION_TYPE_INDEX = 1
+    __TIMESTAMP_INDEX = 2
+    __CURRENCY_INDEX = 3
+    __ASSET_AMOUNT_INDEX = 4
+    __COST_BASIS_INDEX = 5
+    __SOLD_CURRENCY_INDEX = 7
+    __SOLD_AMOUNT_INDEX = 8
+    __SOLD_PROCEEDS_INDEX = 9
+
+    __DELIMITER = ","
+
+    def __init__(
+        self,
+        account_holder: str,
+        transaction_csv_file: str,
+        native_fiat: Optional[str] = None,
+    ) -> None:
+
+        super().__init__(account_holder=account_holder, native_fiat=native_fiat)
+        self.__transaction_csv_file: str = transaction_csv_file
+        self.__logger: logging.Logger = create_logger(f"{self.__COINBASE}/{self.account_holder}")
+
+    def load(self) -> List[AbstractTransaction]:
+        result: List[AbstractTransaction] = []
+
+        with open(self.__transaction_csv_file, encoding="utf-8") as transaction_csv_file:
+            # read CSV with header and skip first row
+            lines = reader(transaction_csv_file)
+
+            # Skip header line
+            header = next(lines)
+            self.__logger.debug("Header: %s", header)
+
+            for line in lines:
+                raw_data: str = self.__DELIMITER.join(line)
+                self.__logger.debug("Transaction: %s", raw_data)
+
+                transaction_id: str = line[self.__TRANSACTION_ID_INDEX].strip()
+                transaction_type: str = line[self.__TRANSACTION_TYPE_INDEX].strip()
+
+                # TODO unsure how to handle conversions, I think this is only for ETH2
```

Comment:
Check how the [REST Coinbase plugin](https://github.com/eprbell/dali-rp2/blob/main/src/dali/plugin/input/rest/coinbase.py) does it (look for "swap" in the code)